From 650f7d76a89625a39749b69371700abe3cd5b1f9 Mon Sep 17 00:00:00 2001
From: Alisher Alikhodjaev <alisher@google.com>
Date: Tue, 31 Oct 2023 11:13:03 -0700
Subject: [PATCH] Possible deadlock on the NfcService object

Bug: 268038643
Bug: 307489565
Test: CtsVerifier
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:2d8ea70c65831313d73784fb3c78f64ff0cbd627)
Merged-In: I683ae425dafa4e209b9517b62ada7d8a694f84a9
Change-Id: I683ae425dafa4e209b9517b62ada7d8a694f84a9
---
 src/com/android/nfc/NfcService.java | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/com/android/nfc/NfcService.java b/src/com/android/nfc/NfcService.java
index f22dc9da..3f860721 100644
--- a/src/com/android/nfc/NfcService.java
+++ b/src/com/android/nfc/NfcService.java
@@ -1304,9 +1304,7 @@ public class NfcService implements DeviceHostListener, ForegroundUtils.Callback
     }
 
     public boolean isSecureNfcEnabled() {
-        synchronized (NfcService.this) {
-            return mIsSecureNfcEnabled;
-        }
+        return mIsSecureNfcEnabled;
     }
 
     final class NfcAdapterService extends INfcAdapter.Stub {
-- 
2.43.0.rc1.413.gea7ed67945-goog

